tikzplotlib/_image.py

```diff
@@ -1,15 +1,53 @@
+from base64 import encodebytes
+from io import BytesIO
+
 import matplotlib.pyplot as plt
 import numpy as np
 import PIL
 
 from . import _files
 
 
+def prepare_image_storage(data):
+    """Prepares a storage location for an image."""
+    if data["embed_images"]:
+        handle = BytesIO()
+        return handle, handle
+    else:
+        filepath, rel_filepath = _files.new_filepath(data, "img", ".png")
+        return filepath, rel_filepath
+
+
+def prepare_image_addplot(data, rel_filepath_or_handle, extent):
+    """Returns an \\addplot command for an image, previously prepared with `prepare_image_storage`."""
+    if not isinstance(rel_filepath_or_handle, BytesIO):
+        # Explicitly use \pgfimage as includegrapics command, as the default
+        # \includegraphics fails unexpectedly in some cases
+        image_embedding_command = "\\pgfimage"
+        image_embedding_parameter = rel_filepath_or_handle
+    else:
+        image_embedding_command = "\\pgfimageembedded"
+        image_embedding_parameter = (
+            "\n" + encodebytes(rel_filepath_or_handle.getbuffer()).decode()
+        )
+
+    # the format specification will only accept tuples
+    if not isinstance(extent, tuple):
```

Comment:
I've moved this check from its former location, but is it ever not a tuple? Or rather: Is it ever something not accessable via indexing?